.github/workflows/upgrade-cdktf.yml

```diff
@@ -0,0 +1,109 @@
+        uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
+      - name: Do the upgrade
+        run: |
+          sed -i "s/cdktfVersion: \".*\",/cdktfVersion: \"^$NEW_CDKTF_VERSION\",/" ./projenrc.template.js
+        env:
+          NEW_CDKTF_VERSION: ${{ needs.check_versions.outputs.latest_version }}
+      - name: Create Pull Request
+        id: cpr
+        uses: peter-evans/create-pull-request@284f54f989303d2699d373481a0cfa13ad5a6666 # v5.0.1
+        with:
+          branch: auto/upgrade-cdktf-${{ needs.check_versions.outputs.latest_version_short }}
+          base: main
+          commit-message: "feat!: upgrade providers to CDKTF version ${{ needs.check_versions.outputs.latest_version }}"
+          title: "feat!: upgrade providers to CDKTF version ${{ needs.check_versions.outputs.latest_version }}"
+          body: This PR upgrades CDKTF from version `${{ needs.check_versions.outputs.current_version }}` to version `${{ needs.check_versions.outputs.latest_version }}` in the prebuilt providers managed by this project.
+          labels: automated
+          token: ${{ secrets.GH_TOKEN_ACTIONS_UPDATER }}
+          author: team-tf-cdk <github-team-tf-cdk@hashicorp.com>
+          committer: team-tf-cdk <github-team-tf-cdk@hashicorp.com>
+          signoff: true
+          delete-branch: true
+    outputs:
+      pr_id: ${{ steps.cpr.outputs.pull-request-number }}
+  upgrade_self:
+    name: Upgrade CDKTF in repository-manager
+    runs-on: ubuntu-latest
+    needs: [check_versions, upgrade_repos]
+    if: needs.check_versions.outputs.current_version_short != needs.check_versions.outputs.latest_version_short
+    permissions:
+      contents: read
+    steps:
+      - name: Checkout
+        uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
+      - name: Install
+        run: yarn install
+      - name: Do the upgrade
+        run: |
+          yarn add cdktf@^$NEW_CDKTF_VERSION
+          yarn add cdktf-cli@^$NEW_CDKTF_VERSION
+        env:
+          NEW_CDKTF_VERSION: ${{ needs.check_versions.outputs.latest_version }}
+      - name: Create Pull Request
+        uses: peter-evans/create-pull-request@284f54f989303d2699d373481a0cfa13ad5a6666 # v5.0.1
+        with:
+          branch: auto/upgrade-cdktf-${{ needs.check_versions.outputs.latest_version_short }}
+          base: main
+          commit-message: "chore: upgrade self to cdktf ${{ needs.check_versions.outputs.latest_version }}"
+          title: "chore: upgrade self to cdktf ${{ needs.check_versions.outputs.latest_version }}"
+          body: |
+            This PR upgrades CDKTF to version `${{ needs.check_versions.outputs.latest_version }}` in `cdktf-repository-manager` (this repo).
+            Unfortunately, not everything can be automated, and the following steps need to be completed manually:
+
+            - [ ] Wait for #${{ needs.upgrade_repos.outputs.pr_id }} to be merged and for the deployment to succeed
```

Comment:
Thinking about this: It would be really cool, if `cdktf provider add` had a flag like `--force-remote` that would make the command fail, if e.g. there's no pre-built provider for Github for the new CDKTF version yet 🤔 Would be simple to implement as well 😄